### DISCUSSION TITLE
Modifier/filter to select every column in a table except one?

### DISCUSSION THREAD

**Discussion by GodGardensGuns** [supabase/supabase#871] [Questions]
I'm building an app where there are multiple possible user types, each with a corresponding table in the database. I'd like to do a dynamic query where I can select all columns in a database for any table (as in `"*"`) but exclude a single column (a `user_id` column in this instance). 

Basically:
```js
const { data, error } = await supabase
        .from(
          `profiles_${userType}`
        )
        // How do I exclude the user_id column here?
        .select("*");
```

Being able to do this would keep me from having to write a huge ternary/switch statement with verbose `.select()` queries that list every column name _except_ `user_id`.

**Comment by Wamy-Dev** [comment]
4 years and still nothing for something seemingly so basic.

**Reply by GaryAustin1** [reply]
Postgres does not have such a feature...
https://dba.stackexchange.com/questions/1957/sql-select-all-columns-except-some

You could use a view or an rpc call to a Postgres function and only return rows you want.

You can't use Postgres column privileges because you then have to specify each column you want as * will fail with an error.
